CONTRIBUTING.md

```diff
@@ -302,12 +302,44 @@ about:
 
 1. [Push your changes] to your GitHub repository.
 
-### Finally...
+### Code review criteria & workflow
+
+At all cases when a Checkbox maintainer picks up a PR for review, they are prepared to go back to the same PR if needed (if changes were requested).
+
+As a general rule there is one reviewer per PR. Other reviewers may pop in to smooth up the process (where they feel confident they can _approve and merge_ changes after some were requested by another review). When more reviewers are needed an approval of, comments explaining that are made in the comment thread. Chiefly this is done to work around cases of low test coverage in cases when the changes affect something known to be of low quality (i.e. something significantly complex and hard to reason about, brittle, dated, known to have broken in the past).
+
+The approving reviewer will merge the code if the review is accepted, and leave a comment if they have for some good reason decided not to merge it at approval time.
+
+The reviewer is encouraged to use suggestions to communicate exact solutions and to make it easy to apply them. The reviewer _must_ do this when making trivial style related suggestions. The reviewer might also post code into the PR, or to a branch branched off from the feature branch, to communicate more complex suggestions. 
+
+Checkbox maintainers will follow the following process to make the determination between "accept", "request changes" and "comment".
+
+### Review will approve the pull request when...
+
+1. They have read and understood the pull request description and changes being proposed.
+2. The requirements laid out [in the PR template](.github/pull_request_template.md) are met. In particular:
+  - the reviewer is convinced the changed code works as advertised
+  - tests introduced cover the new functionality, as well as untouched code it may affect
+  - testing reported by the author covers the new functionality, as well as untouched code it may affect 
+  - if needed, reviewer has tested the changed solution locally
+
+#### Reviewer will "request changes" to your PR when...
+
```

Comment:
```suggestion

1. The PR is too big and should be split in different PRs.
```

Reply:
Will apply this but numbering other items needs fixing.